### PR TITLE
Node 18+ Updates

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Fixes:** #20, #21, #22, #23

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [ ] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

### CODE REVIEW

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Please see the following additional updates needed

1. Update ./.github/settings.yml
2. rm ./karma.conf.js
3. rm -rf ./demo./sass
4. Update all references of unpkg to use JSdelivr
5. Update to Chalk v5 and update all files using it
6. rm index.html
7. Update all dependencies 
8. Remove babel-loader dependency 

Screen Shot 2023-06-15 at 1 47 37 PM

File: package.json [hunk 1]
```diff
@@ -7,14 +7,17 @@
     "  3. Go to http://localhost:8000",
     "================================================================================"
   ],
-  "name": "@alaskaairux/auro-datetime",
+  "name": "@aurodesignsystem/auro-datetime",
```
Comment (blackfalcon) [inline package.json:10]: This is a BREAKING CHANGE

File: index.js [hunk 2]
```diff
@@ -1 +1,16 @@
...
+ */
+export function registerComponent(name) {
+  // alias definition
+  if (!customElements.get(name)) {
+    customElements.define(name, class extends AuroDatetime {});
+  }
+}
+
+// Example custom registration
+// registerComponent('my-badge');
```
Comment (blackfalcon) [inline index.js:16]: Watch that copy pasta

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json [hunk 3]
```diff
@@ -41,9 +44,6 @@
     "@commitlint/config-conventional": "^12.1.1",
     "@open-wc/testing": "^2.5.32",
     "@open-wc/testing-karma": "^4.0.9",
```
Comment (blackfalcon) [inline package.json:46]: Remove this dependency and `karma.conf.js`

File: package.json [hunk 4]
```diff
@@ -59,7 +59,7 @@
     "eslint": "^7.23.0",
     "eslint-plugin-lit": "^1.3.0",
     "focus-visible": "^5.2.0",
-    "husky": "^4.3.8",
+    "husky": "^8.0.3",
```
Comment (blackfalcon) [inline package.json:62]: Be sure all these steps have been completed. Tested locally and pre-commit hook is not firing. 

https://github.com/AlaskaAirlines/auro-badge/issues/43

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 5]
```diff
@@ -0,0 +1,56 @@
+name: Bug report
+description: Create a report to help us improve
+title: '[issue summary] Please verify version before submitting new issue'
+labels:
+  - 'Type: Bug'
+  - not-reviewed
+  - auro-badge
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:7]: Watch that copy and paste... there are multiple references to `badge` in this repo.

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    attributes:
+      value: >
+        [![See it on
+        NPM!](https://img.shields.io/npm/v/@alaskaairux/auro-badge?style=for-the-badge&color=orange)](https://www.npmjs.com/package/@alaskaairux/auro-badge)
+  - type: textarea
+    id: details
+    attributes:
+      label: Please describe the bug
+      description: List out the steps to reproduce the behavior
+      placeholder: 1. Go to '...' 2. Click on '....' 3. Scroll down to '....' 4. See error
```
Comment (blackfalcon) [inline .github/ISSUE_TEMPLATE/bug_report.yml:27]: This is the wrong version of this, check the Generator repo.

File: .github/settings.yml [hunk 7]
```diff
@@ -56,151 +56,106 @@ branches:
...
-  aliases:
-  - refactor
-- name: 'Type: Testing'
-  color: fff0cd
-  description: Adding missing tests or correcting existing tests
-  process: Author applies label type when reporting issue
-  aliases:
-  - test
-  - testing
+  - name: duplicate
```
Comment (blackfalcon) [inline .github/settings.yml:67]: Missing reference to `auro-datetime`, look at the file in the generator.

File: package.json [hunk 8]
```diff
@@ -87,47 +81,41 @@
...
     "demo:copyIndex": "copyfiles -u 1 -V './demo/index.html' ./build",
     "demo:updateIndex": "node ./scripts/prepForBuild",
-    "demo:build": "npm-run-all ciBuild demo:rmBuild demo:newBuild demo:copyIndex demo:copyDemo demo:updateIndex"
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "npm-run-all test lint bundler postinstall",
-      "commit-msg": "commitlint -E HUSKY_GIT_PARAMS"
-    }
+    "demo:build": "npm-run-all ciBuild demo:rmBuild demo:newBuild demo:copyIndex demo:copyDemo demo:updateIndex",
```
Comment (blackfalcon) [inline package.json:116]: Remove all references to demo build

File: scripts/pre-commit.mjs [hunk 9]
```diff
@@ -0,0 +1,17 @@
+'use strict';
```
Comment (blackfalcon) [inline scripts/pre-commit.mjs:1]: Pre-commit hooks are not firing. This file will fail when hooks fire.

**Review by blackfalcon — CHANGES_REQUESTED** [review]
.npmignore and .gitignore are not updated.

File: scripts/pre-commit.mjs (hunk 9 above)
Comment (jordanjones243) [reply scripts/pre-commit.mjs:1]: Jason and I are pair programming now and are struggling to figure out why the pre-commit hooks are failing. Would you have an idea why?

Comment (jordanjones243) [reply scripts/pre-commit.mjs:1]: Pre-commit hooks are now firing, nothing happened with this file.

File: .github/settings.yml [hunk 10]
```diff
@@ -3,13 +3,13 @@ repository:
...
   # A short description of the repository that will show up on GitHub
-  description: HTML custom element that has a built-in date/time API
+  description: Custom element that has a built-in date/time API
 
   # A URL with more information about the repository
   homepage: https://auro.alaskaair.com/components/auro/datetime
 
   # A comma-separated list of topics to set on the repository
-  topics: auro, design-system, custom-element
+  topics: auro, design-system, custom-element, generator-v[genVersionDash]
```
Comment (blackfalcon) [inline .github/settings.yml:12]:
```suggestion
  topics: auro, design-system, custom-element
```

File: .github/settings.yml [hunk 11]
```diff
@@ -56,151 +60,115 @@ branches:
         # Required. Require branches to be up to date before merging.
         strict: true
         # Required. The list of status checks to require in order to merge into this branch
-        contexts: ["test (11.x)", "test (12.x)", "test (14.x)", "license/cla"]
+        contexts: ["test ([abstractNodeVersion])", "license/cla"]
```
Comment (blackfalcon) [inline .github/settings.yml:63]:
```suggestion
        contexts: ["test (18.x)", "license/cla"]
```

File: .gitignore [hunk 12]
```diff
@@ -16,5 +12,5 @@ src/*.css
 dist/
 build/
 
-## Karma code coverage reports
+## Code coverage reports
```
Comment (blackfalcon) [inline .gitignore:15]: This file is wrong, see https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/.gitignore.temp

File: package.json [hunk 13]
```diff
@@ -17,16 +17,18 @@
   },
   "main": "index.js",
   "license": "Apache-2.0",
+  "engines": {
+    "node": ">=18.15.0"
+  },
   "dependencies": {
     "chalk": "^4.1.0",
```
Comment (blackfalcon) [inline package.json:24]: Chalk needs to be updated

File: package.json [hunk 14]
```diff
@@ -17,16 +17,18 @@
...
-    "lit-element": "^2.4.0"
+    "lit": "^2.7.5"
   },
   "peerDependencies": {
     "@alaskaairux/design-tokens": "^3.9.0",
     "@alaskaairux/icons": "^4.13.0",
     "@alaskaairux/webcorestylesheets": "^3.7.3",
-    "@webcomponents/webcomponentsjs": "^2.5.0",
-    "focus-visible": "^5.2.0"
+    "@webcomponents/webcomponentsjs": "^2.5.0"
```
Comment (blackfalcon) [inline package.json:31]: Review generator template, I don't think we are using this.

File: package.json [hunk 15]
```diff
@@ -87,47 +81,35 @@
...
     "last 2 iOS major versions",
-    "ie 11",
     "last 2 Firefox versions",
     "last 2 Edge versions",
     "last 2 Safari major versions"
   ],
   "scripts": {
-    "build": "npm run ciBuild",
-    "apiBuild": "wca analyze 'src/auro-datetime.js' --outFiles docs/api.md",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api",
```
Comment (blackfalcon) [inline package.json:89]: There are a lot of legacy dependencies in here that should be removed. Please review the generator template.

File: src/auro-datetime.js [hunk 16]
```diff
@@ -4,11 +4,11 @@
 // ---------------------------------------------------------------------
 
 // If use litElement base class
-import { LitElement, html } from "lit-element";
+import { LitElement, html } from "lit";
```
Comment (blackfalcon) [inline src/auro-datetime.js:7]:
```
/Users/dalesande/src/auro/datetime/src/auro-datetime.js
  179:77  warning  Expected line break before `.replace`  newline-per-chained-call
  179:97  warning  Expected line break before `.replace`  newline-per-chained-call
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 12 above)
Comment (jordanjones243) [reply .gitignore:21]: Fixed.

File: package.json (hunk 15 above)
Comment (jordanjones243) [reply package.json:80]: Done.

File: package.json [hunk 17]
```diff
@@ -17,117 +17,90 @@
...
+    "build:watch": "nodemon -e scss,js --watch src --exec npm run build:dev:assets",
     "bundler": "rollup -c",
     "bundle:test": "rollup -c -w",
-    "postinstall": "node packageScripts/postinstall.js",
+    "postinstall": "node packageScripts/postinstall.mjs",
     "cssLint": "stylelint \"./src/*.css\"",
-    "dev": "npm run sassBuild:watch",
-    "distJS": "copyfiles -u 1 -V './src/**/*.js' ./dist",
-    "sassRender": "sass-render src/*.css -t ./scripts/staticStyles-template.js",
+    "dev": "concurrently --kill-others 'npm run build:watch' 'npm run serve'",
```
Comment (blackfalcon) [inline package.json:88]: You are missing the `concurrently` package needed to run this. Have you run this locally?

File: package.json [hunk 18]
```diff
@@ -17,117 +17,90 @@
...
+    "chalk": "^5.2.0",
+    "lit": "^2.7.5"
   },
   "peerDependencies": {
     "@alaskaairux/design-tokens": "^3.9.0",
     "@alaskaairux/icons": "^4.13.0",
-    "@alaskaairux/webcorestylesheets": "^3.7.3",
-    "@webcomponents/webcomponentsjs": "^2.5.0",
-    "focus-visible": "^5.2.0"
+    "@alaskaairux/webcorestylesheets": "^3.7.3"
```
Comment (blackfalcon) [inline package.json:30]: This package is deprecated. Install `@aurodesignsystem` version.

File: demo/index.html [hunk 19]
```diff
@@ -25,7 +25,7 @@
           Prism.highlightAll();
         });
     </script>
-    <script type="module" src="../src/auro-datetime.js"></script>
+    <script type="module" src="../index.js"></script>
```
Comment (blackfalcon) [inline demo/index.html:28]: Please reference https://github.com/AlaskaAirlines/WC-Generator/blob/master/template/demo/index.html for all other changes.

File: src/style.scss [hunk 20]
```diff
@@ -4,11 +4,11 @@
...
 // Import Auro tokens
-@import './node_modules/@alaskaairux/design-tokens/dist/tokens/SCSSVariables';
-@import './node_modules/@alaskaairux/design-tokens/dist/tokens/SassCustomProperties';
+@import './../node_modules/@alaskaairux/design-tokens/dist/tokens/SCSSVariables';
+@import './../node_modules/@alaskaairux/design-tokens/dist/tokens/SassCustomProperties';
 
-@import './node_modules/@alaskaairux/webcorestylesheets/dist/breakpoints';
-@import './node_modules/@alaskaairux/webcorestylesheets/dist/core';
+@import './../node_modules/@alaskaairux/webcorestylesheets/dist/breakpoints';
+@import './../node_modules/@alaskaairux/webcorestylesheets/dist/core';
```
Comment (blackfalcon) [inline src/style.scss:11]: Update to use new `@aurodesignsystem` versions of the resources.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Update package name to new @aurodesignsystem npm namespace. 

Also, I saw the following warning...

```
hint: The '.husky/commit-msg' hook was ignored because it's not set as executable.
```

The fix is to run the following 

```
chmod ug+x .husky/*
```

It's in the last commit. 

See other notes for requested updates.

**Comment by jordanjones243** [comment]
> 

Update the package name of the entire repo to @AuroDesignSystem ?

File: package.json (hunk 1 above)
Comment (blackfalcon) [reply package.json:10]: Please update the comment for this commit in the body to include `BREAKING CHANGE`. Also, due to this namespace change, this CANNOT be shipped until the repo secrets are updated too. Once this work is approved I will update the secrets.

**Review by blackfalcon — CHANGES_REQUESTED** [review]

File: package.json [hunk 21]
```diff
@@ -17,117 +17,91 @@
...
     "last 2 iOS major versions",
-    "ie 11",
     "last 2 Firefox versions",
     "last 2 Edge versions",
     "last 2 Safari major versions"
   ],
   "scripts": {
-    "build": "npm run ciBuild",
-    "apiBuild": "wca analyze 'src/auro-datetime.js' --outFiles docs/api.md",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api",
```
Comment (blackfalcon) [inline package.json:81]:
```suggestion
    "build": "npm-run-all build:sass sass:render scssLint dist:js bundler postinstall build:api types build:docs",
```

File: package.json [hunk 22]
```diff
@@ -17,117 +17,91 @@
...
+    "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
+    "build:api": "wca analyze 'src/auro-datetime.js' --outFiles docs/api.md",
+    "build:watch": "nodemon -e scss,js --watch src --exec npm run build:dev:assets",
     "bundler": "rollup -c",
     "bundle:test": "rollup -c -w",
-    "postinstall": "node packageScripts/postinstall.js",
+    "postinstall": "node packageScripts/postinstall.mjs",
     "cssLint": "stylelint \"./src/*.css\"",
```
Comment (blackfalcon) [inline package.json:88]:
```suggestion
    "scssLint": "stylelint \"./src/**/*.scss\"",
```

File: package.json [hunk 23]
```diff
@@ -17,117 +17,91 @@
...
-    "test": "karma start --coverage",
-    "test:watch": "karma start --auto-watch=true --single-run=false",
-    "test:ci": "npm-run-all test lint",
-    "ciBuild": "npm-run-all sassBuild sassRender cssLint distJS bundler postinstall apiBuild",
-    "sassBuild": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:dev": "npm-run-all sassBuild:demo sassBuild:component postCss:component sassRender",
-    "sassBuild:demo": "node-sass ./demo/sass/ --output ./demo/css/",
-    "sassBuild:component": "node-sass ./src/ --output ./src/",
-    "sassBuild:watch": "nodemon -e scss --watch src --exec npm run sassBuild:dev",
+    "linters": "npm-run-all cssLint esLint",
```
Comment (blackfalcon) [inline package.json:94]:
```suggestion
    "linters": "npm-run-all scssLint esLint",
```

File: package.json [hunk 24]
```diff
@@ -17,117 +17,91 @@
...
-    "ie 11",
     "last 2 Firefox versions",
     "last 2 Edge versions",
     "last 2 Safari major versions"
   ],
   "scripts": {
-    "build": "npm run ciBuild",
-    "apiBuild": "wca analyze 'src/auro-datetime.js' --outFiles docs/api.md",
+    "build": "npm-run-all build:sass sass:render cssLint dist:js bundler postinstall build:api",
+    "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
```
Comment (blackfalcon) [inline package.json:82]:
```suggestion
    "build:dev:assets": "npm-run-all build:sass:component postCss:component sass:render",
    "build:docs": "node scripts/generateDocs.js",
```

**Review by blackfalcon — CHANGES_REQUESTED** [review]
Also update all dependencies 

```
 Patch Update Backwards-compatible bug fixes.
❯◯ lit                2.7.5   ❯  2.7.6   https://lit.dev/
 ◯ core-js devDep     3.31.0  ❯  3.31.1  https://github.com/zloirock/core-js#readme
 ◯ postcss devDep     8.4.24  ❯  8.4.25  https://postcss.org/
 ◯ typescript devDep  5.1.3   ❯  5.1.6   https://www.typescriptlang.org/

 Minor Update New backwards-compatible features.
 ◯ chalk             5.2.0   ❯  5.3.0    https://github.com/chalk/chalk#readme
 ◯ eslint devDep     8.43.0  ❯  8.44.0   https://eslint.org
 ◯ stylelint devDep  15.9.0  ❯  15.10.1  https://stylelint.io

 Major Update Potentially breaking API changes. Use caution.
 ◯ @semantic-release/changelog devDep  5.0.1   ❯  6.0.3   https://github.com/semantic-release/changelog#readme
 ◯ @semantic-release/git devDep        9.0.1   ❯  10.0.1  https://github.com/semantic-release/git#readme
 ◯ @semantic-release/npm devDep        7.1.3   ❯  10.0.4  https://github.com/semantic-release/npm#readme
 ◯ nodemon devDep                      2.0.22  ❯  3.0.1   https://nodemon.io
 ◯ rollup devDep                       2.79.1  ❯  3.26.2  https://rollupjs.org/
 ◯ rollup-plugin-serve devDep          1.1.0   ❯  2.0.2   https://github.com/thgh/rollup-plugin-serve
 ◯ semantic-release devDep             17.4.7  ❯  21.0.7  https://github.com/semantic-release/semantic-release#readme
 ◯ stylelint-config-standard devDep    33.0.0  ❯  34.0.0  https://github.com/stylelint/stylelint-config-standard#readme

 Non-Semver Versions less than 1.0.0, caution.
 ◯ @web/dev-server devDep  0.2.3  ❯  0.2.5  https://github.com/modernweb-dev/web/tree/master/packages/dev-server
```

**Review by blackfalcon — APPROVED** [review]

**Pull request merged**